### ISSUE TITLE
Language list changed mail still in the default language

### ISSUE THREAD

**Issue by bertaviader** [sympa-community/sympa#573]
Hello,

We have a Sympa version 6.2.24. In the sympa.conf the configuration for the language is the next one:

lang    ca
supported_lang es,ca,en-US

In a list we change de Language of the list (lang) to english and in the config is displayed like this: 

> lang en

On the website the options are displayed like this:
![Captura de 2019-03-26 11-46-07](https://user-images.githubusercontent.com/18306366/54991422-377f1780-4fbd-11e9-9b78-9fcc03b60cab.png)

The expected result if I change the language is that all the communication in that list, for example the action to subscribe or unsubscribe will be in English, but they are in Catalan, the default language.
We found that changing the configuration in the list to: lang en-US and putting the language on the website in English. The communication are set in the correct language. It happens the same in Spanish, but we couldn't figure out which is the correct form of the config in Spanish. 

So there are so thinks to take into account, the wrong saving of the language when we change it and that the website needs to be in the same language as the list to change the language of the list.

Can you help us on this?

Thank you,
Berta

**Comment by dverdin** [comment]
If a user has set a language in preferences, this language will be used, taking precedence over the lists settings (if the language is enabled in the configuration).
Regards,
David

**Comment by bertaviader** [comment]
Thanks we didn't know that. But I think the problem is, I'm new to sympa, no entry in the user table, someone subscribes me to a list, this list is in English. So in the database my language it's set to English. I don't do anything in sympa. Then someone subscribes me to another list in Catalan. The message are still in English. How can we force that if you don't select a language the language will be defined by the language list?

Thanks,
Berta

**Comment by dverdin** [comment]
I see your problem.
And the default language _for the server_ is Catalan ? Not English?
I think Sympa should probably not set a language until the user specifically chooses one.
It might be an improvement to ask.

**Comment by racke** [comment]
And it would be certainly useful to document how the language is determined.

**Comment by ikedas** [comment]
> And it would be certainly useful to document how the language is determined.

Yes, it is [wanted](https://sympa-community.github.io/manual/customize/basics-i18n.html#sympa-internationalization-1) :wink:.

**Comment by dverdin** [comment]
Work in progress indeed.
shall I consider that it would be nice if, for example, _I_ wrote something about it? :smirk:

**Comment by ikedas** [comment]
Of course.
But we at first have to know how Sympa works exactly with language settings. I wonder it sometimes looks behaving a bit weird,

**Comment by dverdin** [comment]
Indeed. I have recurring issues about the exact same situation posted here.
That requires a bit of research in the code, I guess.

**Comment by racke** [comment]
The code is in `Sympa::best_language` and `Sympa::Language::negotiate_lang`. It takes the browser settings into the account which might lead to the confusion.

**Comment by bertaviader** [comment]
The strange thing is that the default language is in Catalan, but if a user has never entered into Sympa, he doesn't have an entry in the user_table, and if the first list I subscribe him it's in English, then the default language for he is English.
In the database there are user entries with the language in NULL, in this case these people receive the mails in the language of the list. We don't know why we have this in NULL, maybe because we had a previous version of Sympa 5.3.4. But it's just a supposition.

**Comment by ikedas** [comment]
@bertaviader, at least with current behavior of Sympa, if English is included in supported languages, it can be chosen.  See #46 (that last comment by me).

**Comment by bertaviader** [comment]
My browser is in Catalan, but the behavior I have explaining in my last post is still happening. I think that the language sympa is taking into account, is the language of the first list you are subscribed, if you don't change it in the sympa portal. I don't want to force a language, I want that if you don't have a default language, Sympa don't force it.

**Comment by racke** [comment]
It would be logically for Sympa to save the language that you are using the first time.

**Comment by bertaviader** [comment]
But as I understand it if I didn't choose for the language, it's logical that Sympa assume my language as the language of the list first subscribed? Or it will be more logical to put a NULL in the database?

**Comment by dverdin** [comment]
I think that it should be a NULL, thus defaulting to the context (either server or list) until you explicitely set preferences.
